[FE]: Dashboard Page UI has been created

drakkomaximo opened the pull request:
### StarShop Dashboard Progress

**Implemented Features:**

1. **Dashboard Layout:**  
   - The main structure for the dashboard is set up using a clean card-based design to display key sections like store information, top products, and sales statistics.

2. **About Your Store:**  
   - A section showing basic details about the store, including the store's name and a brief description.

3. **Top Products:**  
   - A section to display the store's featured or top-selling products.

4. **Store Statistics:**  
   - A section dedicated to key metrics showing the store's performance, like sales data.

5. **Monthly Sales:**  
   - A section to show the store's sales performance for each month.

6. **Responsive Design:**  
   - The layout adjusts well to different screen sizes, from mobile to desktop, ensuring a good user experience on all devices.

7. **Reusable Components:**  
   - Using a component-based design to keep the dashboard flexible and easy to update.

8. **Centered Content:**  
   - Ensured the content is always centered and doesn’t stretch beyond a maximum width of 1440px for consistency.

**What to Verify:**
- Check how the dashboard looks on different screen sizes.
- Make sure images load correctly.
- Test all links and buttons to ensure they work properly.
- Ensure text is easy to read and has good contrast.


![image](https://github.com/user-attachments/assets/df398919-36ee-4f12-bbc7-4f075e7a0639)
![image](https://github.com/user-attachments/assets/c4f9e428-2147-491f-a63d-d5537ec08f51)

KevinLatino commented on src/components/dashboard/DashboardComponent.tsx:
```diff
@@ -1,10 +1,73 @@
-export default DashboardComponent
\ No newline at end of file
+  return (
+    <div className="min-h-screen space-y-24 bg-[#1a1a2e] text-white p-4 sm:p-6 lg:p-8">
```
```suggestion
    <div className="min-h-screen space-y-24 text-white p-4 sm:p-6 lg:p-8">
```
please, delete the bg-color

drakkomaximo replied:
Done

KevinLatino commented on src/components/dashboard/components/CardContainer.tsx:
```diff
@@ -0,0 +1,85 @@
+};
+
+// Attach subcomponents to CardContainer
+CardContainer.Header = Header;
```
Please, delete subcomponents, and all it's implementations

drakkomaximo replied:
Done

drakkomaximo commented:
# Summary of Changes

## 1. **CardContainer Adjustments**
   - Refactored `CardContainer` from a functional wrapper to a normal component.
   - Handled sections like "Sells Per Month" differently in the component hierarchy.

## 2. **Dashboard Component Layout**
   - Updated the layout of `DashboardComponent` to use the new structure of `CardContainer`.
   - Adjusted child components such as `TopProducts`, `AboutStore`, `Statistics`, and `SellsPerMonth`.
   - Reorganized sections in the `DashboardComponent` to ensure a responsive grid layout.

## 3. **Chart Update - SellsPerMonth**
   - Changed the chart type from a line chart to a donut chart (circular bar chart).
   - Displayed five specific metrics: "Trustless Work", "SotNet", "Safetrust", "ScanGuard", and "Other".
   - Selected custom colors for each section of the donut chart.

These updates improve the component structure, UI elements, and responsiveness across the application.

KevinLatino commented on src/components/dashboard/components/CardContainer.tsx:
```diff
@@ -0,0 +1,57 @@
+  footer
+}) => {
+  return (
+    <div className={`flex flex-col justify-end ${className}`}>
+      {/* Header Section */}
+      {(icon || title) && (
+        <div
+          className={`flex gap-2 items-center mb-2 pl-4 ${
+            !icon && !title ? "h-[44px]" : ""
+          }`}
+        >
+          {icon && <div>{icon}</div>}
+          {title && (
+            <h3 className="text-[20px] sm:text-[24px] lg:text-[30px] font-bold text-center text-background">
+              {title}
+            </h3>
+          )}
+        </div>
+      )}
+
+      {/* Body Section */}
+      <div className="flex flex-col items-center px-4 pb-4 border-4 border-custom-light-card-border rounded-[20px] sm:rounded-[24px] lg:rounded-[28px] max-h-[254px] h-screen">
+        {subtitle && (
+          <h4 className="text-center text-lg sm:text-xl lg:text-2xl font-semibold mt-2">
+            {subtitle}
+          </h4>
+        )}
+        {children}
+      </div>
+
+      {/* Footer Section */}
+      {footer && <div className="mt-4">{footer}</div>}
+    </div>
```
```suggestion
    <div className={`flex flex-col justify-end ${className}`}>
      {/* Header Section */}
      {(icon || title) && (
        <div
          className={`flex gap-2 items-center mb-2 pl-4 ${
            !icon && !title ? "h-[44px]" : ""
          }`}
        >
          {icon && <div>{icon}</div>}
          {title && (
            <h3 className="text-[20px] sm:text-[24px] lg:text-[30px] font-bold text-center text-background">
              {title}
            </h3>
          )}
        </div>
      )}

      {/* Body Section */}
      <div className="flex flex-col items-center px-4 pb-4 border-4 border-custom-light-card-border rounded-[20px] sm:rounded-[24px] lg:rounded-[28px] max-h-[254px] h-screen">
        {subtitle && (
          <h4 className="text-center text-lg sm:text-xl lg:text-2xl font-semibold mt-2">
            {subtitle}
          </h4>
        )}
        {children}
      </div>

      {/* Footer Section */}
      {footer && <div className="mt-4">{footer}</div>}
    </div>
```
Make this more flexible, don't use an estructure (title, icon, footer) just use children prop

drakkomaximo replied:
The behavior of the CardContainer was adjusted to make it a simple container.

KevinLatino reviewed (changes requested):
Please delete all the comments

drakkomaximo commented:
![image](https://github.com/user-attachments/assets/d0fea453-6970-4148-9ce4-993d9169bddc)
